File: FrontOffice/src/app/entities/LigneCarnet.ts
export class LigneCarnet {
  prix_achat: number;
  quantite_achat: number;
  nb_ordres_achat: number;
  prix_vente: number;
  quantite_vente: number;
  nb_ordres_vente: number;

  constructor(
    prix_achat: number,
    quantite_achat: number,
    nb_ordres_achat: number,
    prix_vente: number,
    quantite_vente: number,
    nb_ordres_vente: number
  ) {
    this.prix_achat = prix_achat;
    this.quantite_achat = quantite_achat;
    this.nb_ordres_achat = nb_ordres_achat;
    this.prix_vente = prix_vente;
    this.quantite_vente = quantite_vente;
    this.nb_ordres_vente = nb_ordres_vente;
  }
}

File: BackEnd/src/main/java/com/example/pifinance_back/Entities/LigneCarnet.java
package com.example.pifinance_back.Entities;

import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class LigneCarnet {
    private double prixAchat;
    private int quantiteAchat;
    private int nbOrdresAchat;
    private double prixVente;
    private int quantiteVente;
    private int nbOrdresVente;

    // Constructeurs, getters et setters ici...

    // Vous pouvez ajouter des méthodes pour ajouter des ordres d'achat ou de vente
    public void ajouterOrdreAchat(OrdreAchat ordre) {
        this.prixAchat = ordre.getPrixLimiteA();  // Assure que le prix de vente est mis à jour
        this.quantiteAchat += ordre.getQuantite();
        this.nbOrdresAchat++;
    }

    public void ajouterOrdreVente(OrdreVente ordre) {
        this.prixVente = ordre.getPrixLimiteV();  // Assure que le prix de vente est mis à jour
        this.quantiteVente += ordre.getQuantite();
        this.nbOrdresVente++;
    }
}
